src/Invoice_GenUI/Invoice_GenUI/ViewModels/InvoiceDetailsViewModel.cs

```diff
@@ -41,7 +41,7 @@ private async Task GetInvoiceID()
             DueDate = singleInvoice.DueDate.Date;
             LineItemDetails = singleInvoice.LineItems;
 
-            foreach (var item in singleInvoice.LineItems)
+            foreach (var item in singleInvoice.LineItems!)
```

Comment:
:pick: should be able to get around having to use the "damnit" operator (`!`) by checking on how the `LineItems` property is declared in your view model.

Reply:
It might be the wrong model (and correct me if it is), but the `LineItems` property here is created as null:

https://github.com/jamie-taylor-rjj/2023-digital-t-level-apprentice/blob/92aa363f57e6b21da9e65647602c248938b62979/src/Invoice_GenUI/Invoice_GenUI/ViewModels/InvoiceDetailsViewModel.cs#L18

If you change this to

``` csharp
public ObservableCollection<LineItemModel>? LineItemDetails { get; set; } = new ObservableCollection<LineItemModel>();
```

You should be able to get around having to use the damnit operator.

The above line uses any passed in `ObservableCollection<LineItemModel>` to set the value of the property LineItemDetails (via the `set;` auto property); but if one isn't passed in, it will create a new `ObservableCollection<LineItemModel>` rather than assign it to null

Reply:
LineItems is in the InvoiceModel, I did it for the LineItemDetails too though